demo/apple/QuranSyncDemo/QuranSyncDemo/DatabaseManager.swift

```diff
@@ -6,36 +6,31 @@
 //
 
 import Foundation
+import KMPNativeCoroutinesAsync
 import Shared
 
 class DatabaseManager {
   static let shared = DatabaseManager()
   
-  private let database: QuranDatabase
+  private let pageBookmarksRepository: PageBookmarksRepository
   
   private init() {
     let driverFactory = DriverFactory()
-    self.database = DriverFactoryKt.makeDatabase(driverFactory: driverFactory)
+    self.pageBookmarksRepository = PageBookmarksRepositoryFactory.companion.createRepository(driverFactory: driverFactory)
   }
   
-  // Provides access to bookmark queries
-  var bookmarkQueries: BookmarksQueries {
-    return database.bookmarksQueries
+  func bookmarksSequence() -> any AsyncSequence<[PageBookmark], Error> {
+    asyncSequence(for: pageBookmarksRepository.getAllBookmarks())
   }
```

Comment:
I think this is where we provide a small wrapper lib on top of this to expose native swift APIs, right? I would also expect this lib to live in mobile-sync-spm/mobile-sync-ios. Is my understanding correct?

Reply:
Ah, I see. Probably here is better since the other repo just have a single package that we can replicate for local testing with a local xcframework. What do you think?

Reply:
agree that here is better, but let's see what ends up being easier from a ci and usability perspective